onnxruntime/core/optimizer/pad_fusion.cc

```diff
@@ -8,25 +8,7 @@
 
 namespace onnxruntime {
 
-/*
- * It matches following pattern:
- *     Pad
- *      |
- *   Conv/MaxPool/AveragePool
- */
-bool PadFusion::SatisfyCondition(const Graph& graph, const Node& node, const logging::Logger&) const {
-  // if Pad has input axis, don't fuse it.
-  if (!graph_utils::IsSupportedOptypeVersionAndDomain(node, "Pad", {1, 2, 11, 13, 18, 19}) ||
-      node.GetOutputEdgesCount() != 1 ||
-      node.InputDefs().size() > 3) {
-    return false;
-  }
-
-  if (graph.NodeProducesGraphOutput(node)) {
-    return false;
-  }
-
-  const Node& child_node = *node.OutputNodesBegin();
+bool VerifyNonCastChild(const Node& child_node) {
```

Comment:
From the POV of this function, the function (AFAICS) does *not* test the child of the given node. Rather, it tests the given node.

```suggestion
bool VerifyNotCastNode(const Node& node) {
```

Also, `non-` is a prefix rather than a word itself (nonsense, nonalcoholic, nonfat, nonexistent), and thus would be `Noncast`.

Reply:
From the point of the given function, the node is not a *child* node. It is simply a node.